src/components/utils/hooks/useMessagePreviews.tsx

```diff
@@ -40,26 +46,59 @@ const useMessagePreviews = () => {
     return parsed.members.find((member) => member !== userProfileId) ?? null;
   };
 
-  const request = { profileIds: Array.from(profileIds.values()) };
-  const { loading: profilesLoading, error: profilesError } = useQuery(ProfilesDocument, {
-    variables: {
-      request: request
-    },
-    skip: profileIds.size === 0 || currentProfile?.id !== selectedProfileId,
-    onCompleted: (data) => {
-      if (!data?.profiles?.items.length) {
-        return;
-      }
-      const profiles = data.profiles.items as Profile[];
+  useEffect(() => {
+    if (profilesLoading) {
+      return;
+    }
+    const toQuery = new Set(profileIds);
+    // Don't both querying for already seen profiles
+    for (const profile of messageProfiles.values()) {
+      toQuery.delete(profile.id);
+    }
+
+    if (!toQuery.size) {
+      return;
+    }
+
+    const loadLatest = async () => {
+      setProfilesLoading(true);
       const newMessageProfiles = new Map(messageProfiles);
-      for (const profile of profiles) {
-        const peerAddress = profile.ownedBy as string;
-        const key = buildConversationKey(peerAddress, buildConversationId(currentProfile?.id, profile.id));
-        newMessageProfiles.set(key, profile);
+      const chunks = chunkArray(Array.from(toQuery), MAX_PROFILES_PER_REQUEST);
+      try {
+        const results = await Promise.all(
+          chunks.map((profileIdChunk) =>
+            apolloClient.query({
+              query: ProfilesDocument,
+              variables: { request: { profileIds: profileIdChunk } }
+            })
```

Comment:
**question:** I think we can use `useLazyQuery` hook here 🤔, if we use `apolloClient` all the apollo config will not be applied here